authors/apps/articles/views.py

```diff
@@ -131,3 +135,85 @@ def put(self, request, slug):
                                            context={'request': request},
                                            partial=True)
         return Response(serializer.data, status.HTTP_200_OK)
+
+
+class AverageRatingsAPIView(APIView):
+    """
+    Class for viewing article ratings
+    """
+
+    serializer_class = RatingSerializer
+
+    def get(self, request, slug):
+        """method for viewing average ratings"""
+        rate_article = None
+        rate = 0
+
+        try:
+            article = Article.objects.get(slug=slug)
+        except ObjectDoesNotExist:
+            raise NotFound('article not found')
+
+        art_id = article.id
+
+        try:
+            rate_articles = ArticleRating.objects.filter(article_id=art_id)
+        except Exception as e:
+            raise serializers.ValidationError(str(e))
+
+        for rate_article in rate_articles:
+            rate += rate_article.rate
+        rate_value = 0
+        if rate:
+            rate_value = rate / len(rate_articles)
+        return Response(
+            data={"ratings": round(rate_value, 1)}, status=status.HTTP_200_OK)
+
+
+class PostRatingsAPIView(CreateAPIView):
+    """
+    Class for rating an article
+    """
+    permission_classes = (IsAuthenticated,)
+    serializer_class = RatingSerializer
+
+    def post(self, request, slug):
+        """method for posting a rating"""
+        data = request.data
+        serializer = self.serializer_class(data=data)
+        serializer.validate(data=data)
```

Comment:
You don't need this to validate the serializer. What is it doing?
Context: https://www.django-rest-framework.org/api-guide/serializers/#validation

Reply:
Let me get rid of it